tools/autograd/gen_python_functions.py

```diff
@@ -1541,3 +886,115 @@ def op_name(declaration):
                 '{}: name ends with \'_out\', expecting output params'.
                 format(declaration['name']))
         return name
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+#
+#                       Codegen API Integration
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+
+# These helper functions allow us to call the new codegen API from the
+# old codegen script (which operates on Declarations.yaml).
+
+# TODO: remove all these HACKs after migration is completed!
+
+# function schema str -> NativeFunction
+NF_TABLE: Optional[Dict[str, NativeFunction]] = None
+
+def init(native_yaml_path: str) -> None:
+    from tools.codegen.gen import parse_native_yaml
+    global NF_TABLE
+    NF_TABLE = {str(f.func): f for f in parse_native_yaml(native_yaml_path)}
+
+# Multiple decl entries can map to the same native function (because of deprecated decl).
+def decl_to_native_function(decl: Dict[str, Any]) -> NativeFunction:
+    assert NF_TABLE is not None, 'need to initialize codegen.api.python with init()'
+    function_schema_str = decl['schema_string'][6:]
```

Comment:
This looks super extra questionable lol (stripping `aten::` I guess?)

Reply:
fixed by explicitly checking and removing 'aten::' prefix.